contracts/Vault.vy

```diff
@@ -149,15 +143,21 @@ def maxMint() -> uint256:
     return MAX_UINT256
 
 
+@view
+@internal
+def _previewMint(sharesAmount: uint256) -> uint256:
+    return ((shareAmount * self.asset.balanceOf(self)) - 1 / self.totalSupply_ + 1
```

Comment:
Ser in Vyper we don't have namespace issues because of `self.`
```suggestion
    return ((shareAmount * self.asset.balanceOf(self)) - 1 / self.totalSupply + 1
```

Reply:
👍